nucypher/blockchain/eth/signers/hardware.py

```diff
@@ -0,0 +1,243 @@
+"""
+ This file is part of nucypher.
+
+ nucypher is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ nucypher is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+
+from collections import namedtuple
+from typing import List, Tuple, Union
+
+import rlp
+import trezorlib
+from eth_account._utils.transactions import assert_valid_fields, Transaction
+from eth_utils.address import to_canonical_address
+from eth_utils.applicators import apply_key_map
+from eth_utils.conversions import to_int
+from functools import wraps
+from hexbytes import HexBytes
+from trezorlib import ethereum
+from trezorlib.client import get_default_client
+from trezorlib.tools import parse_path, Address, H_
+from trezorlib.transport import TransportException
+from web3 import Web3
+
+from nucypher.blockchain.eth.decorators import validate_checksum_address
+from nucypher.blockchain.eth.signers.base import Signer
+
+
+def handle_trezor_call(device_func):
+    """
+    Decorator for calls to trezorlib that require physical device interactions.
+    """
+    @wraps(device_func)
+    def wrapped(trezor, *args, **kwargs):
+        import usb1  # may not be installable on some systems (consider CI)
```

Comment:
I definitely wrote some mocks and things for this specifically to prevent CI errors. Can you not find them?

Reply:
Yeah - I am having a hard time recreating that exact construction - but I am certainly interested in reviving it - perhaps you can lend a hand?